### PR TITLE
[flang][openacc] Lower init and shutdown directive

### PR DESCRIPTION
**Pull request by clementval**

### CODE REVIEW

**Review by schweitzpgi — APPROVED** [review]
Looking good. Can you take care of the conflicts? Thank you.

**Comment by clementval** [comment]
@schweitzpgi Conflicts have been resolved. Might have new conflict after merging #526 and #528

**Comment by clementval** [comment]
Looks like there is conflict. Can be merged when you want @schweitzpgi

**Pull request merged**
